### ISSUE TITLE
TextEditor: Accept "file:line:col" argument format

### ISSUE THREAD

**Issue by BenWiederhake** [SerenityOS/serenity#5589]
Many tools like g++, grep, shellcheck, etc. print file locations as file:line:col, for example `Meta/CMake/utils.cmake:89` or:

```
../Kernel/Net/TCPSocket.cpp:245:34: error: 'struct Kernel::TCPSocket::OutgoingPacket' has no member named 'tx_timexx'; did you mean 'tx_time'?
```

Many editors like QtCreator, Geany, or CLion understand this format, and open the file with the text cursor ready to go at the desired location. b474f4916479fbf64e6fb5a66cb25b8496e153b3 inspired me to request that our TextEditor (and maybe HackStudio?) should do the same. :)

Maybe this could be a LibCore feature? Dunno.

CC @ry755

**Comment by tomuta** [comment]
What if the filename contains a colon? It's a valid file name character.

**Comment by ry755** [comment]
What if there was a command line option that enables this behavior, and otherwise the colon would be considered part of the filename? (possibly replacing the command line option added in b474f4916479fbf64e6fb5a66cb25b8496e153b3)
For example, Visual Studio Code has this option:
`-g --goto <file:line[:character]> Open a file at the path on the specified line and character position.`
`code -g example.txt:2` will open `example.txt` to line 2, while `code example.txt:2` will open a file called `example.txt:2`

Something similar could be used here in case the user wants colons to be interpreted this way, and the default behavior would stay how it currently is.

**Comment by ElDonad** [comment]
Hello there ! 

Yeah, I thought of that solution too. I settled initially for the query parameters because the LaunchServer uses file extension to infer which application to use, and we would need to make extra checks to ensure the line number isn't mistaken for part of the extension, when query parameters were already sanitized. But either way would be totally fine, I just took the path of least resistance 'cause I'm lazy...

My assumption is that HackStudio probably shouldn't rely on the LaunchServer to open files, since I assume we want them to be opened in HackStudio whatever the default behaviour is. 

Also the Terminal relies on a hidden "hyperlink", at least to know where links point to, and since gcc and the like probably don't do that, either we could make them do it, or we could implement some sort of custom pattern detection in the HackStudio terminal to handle this kind of links. Or something else, that's all I can think of at the moment.

As for the system Terminal, I'm not sure what to think...
